### PR TITLE
Elaborate on C# Connect/Disconnect

### PR DESCRIPTION
**Pull request by 31**
The C# signal page mentions the `Connect` and `Disconnect` APIs (defined on `GodotObject`), but not when you might want to use them, or how. When looking at the class doc, there's technically an example in there, but it seems limited by only translating the code examples rather than having a C#-specific description. It's never mentioned why you might still want/need to use these. I added some examples and some details about this.

This also clarifies the note about https://github.com/godotengine/godot/issues/70414. Right now it doesn't link to the active issue, so I added a link, and it also doesn't mention that using `.Connect` is another way around it. Depending on how much a particular dev hates manual cleanup, this might be significantly more appealing.

I imagine the note might have been kept brief rather than mentioning more about it because this was imagined at the time to perhaps be a short-lived problem. But there doesn't seem to me to be a clear, consensus path forward at https://github.com/godotengine/godot/pull/73730, so I think it's worth the detail.

/cc @paulloz (https://github.com/godotengine/godot-docs/pull/8538)

### CODE REVIEW

File: tutorials/scripting/c_sharp/c_sharp_signals.rst
Comment (paulloz) [inline tutorials/scripting/c_sharp/c_sharp_signals.rst:39]: The "as events" on the line above already implies the "connected using `+=`" for me, that's why it was worded like this.

Comment (AThousandShips) [inline tutorials/scripting/c_sharp/c_sharp_signals.rst:39]:
```suggestion
    signals connected using ``+=`` aren't. Meaning that you will need to manually disconnect (using ``-=``)
```

File: tutorials/scripting/c_sharp/c_sharp_signals.rst
Comment (paulloz) [inline tutorials/scripting/c_sharp/c_sharp_signals.rst:45]: When I did the other PR, we elected to not mention/link bug issues in the docs.

File: tutorials/scripting/cross_language_scripting.rst
Comment (paulloz) [inline tutorials/scripting/cross_language_scripting.rst:45]: Signal delegates need to end with `EventHandler`.
```suggestion
        [Signal] public delegate void MySignalEventHandler();
```

File: tutorials/scripting/c_sharp/c_sharp_signals.rst
Comment (31) [reply tutorials/scripting/c_sharp/c_sharp_signals.rst]: I think this is ambiguous:

> While all engine signals connected as events are automatically disconnected when nodes are freed, custom signals aren't.

While all (engine signals connected as events) are automatically disconnected [...], (custom signals) aren't.
vs.
While all (engine signals) (connected as events) are automatically disconnected [...], (custom signals) (connected as events [implied]) aren't.

I don't know if that's enough to convey what I mean, but I didn't even consider the second interpretation--I thought this line was entirely mistaken when I first read it. 😄

Comment (paulloz) [reply tutorials/scripting/c_sharp/c_sharp_signals.rst]: Yeah, the whole sentence is too complicated for me either way, so I don't really have a strong opinion. I hate we don't have a simple way to convey `+=`.

Comment (31) [reply tutorials/scripting/c_sharp/c_sharp_signals.rst]: To me, without the `:`, it's a conversational sentence fragment, but it doesn't read all that well with the `:`. Maybe `Meaning:` would sound ok to me. I think `This means` is safe/natural/boring (and pushed that for now), but I'm open to whatever.

Comment (31) [reply tutorials/scripting/c_sharp/c_sharp_signals.rst]: I'll take another look later with fresh eyes... yeah, pulling apart the complexity a bit is probably the right move.

And yeah, this is all already a nightmare in casual convos, getting it right in a doc seems like it may be tough. 😅

**Review by raulsntos — APPROVED** [review]
As you say I did imagine it wouldn't take too long to fix and didn't want to discourage users from the event syntax. Unfortunately, there's problems with the proposed fix (https://github.com/godotengine/godot/pull/73730) so it may still take a while. Therefore, I'm more than willing to correct the documentation now (as it should've been done long ago).

Thank you for contributing this PR, it looks good to me so I'll approve but let's wait for @paulloz approval too.

**Comment by paulloz** [comment]
It’s fine by me. And (re)documenting the calls to `Connect` and `Disconnect` is a good thing.

>

**Review by skyace65 — APPROVED** [review]

**Comment by skyace65** [comment]
Thanks!

File: tutorials/scripting/c_sharp/c_sharp_signals.rst
Comment (31) [reply tutorials/scripting/c_sharp/c_sharp_signals.rst]: After playing with the behavior a bit, I found a whole new chapter to this about capturing variables, so ended up writing a decent amount at https://github.com/godotengine/godot-docs/pull/8815.

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.2 in https://github.com/godotengine/godot-docs/pull/9648.

**Pull request merged**
